### PR TITLE
[Whisper Base] Add two metrics to measure non real-time performance

### PR DESCRIPTION
**Pull request by Honry**
- TTFT: time to first token
- TPS: token per second

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @huningxin, @ibelem, PTAL, thanks!

Preview:
image

File: demos/whisper-base/main.js [hunk 1]
```diff
@@ -105,6 +105,9 @@ let subAudioChunkLength = 0; // length of a sub audio chunk
 let subText = "";
 let speechToText = "";
 
+let time_to_first_token = 0; // TTFT
+let num_tokens = 0; // number of tokens
```
Comment (huningxin) [inline demos/whisper-base/main.js:109]: Do we need to reset the num_tokens to 0 for a new inference session? Or I may miss something.

Comment (Honry) [reply demos/whisper-base/main.js:109]: Thanks, good catch, `num_tokens` should be reset.

Fixed and PTAL again.

File: demos/whisper-base/main.js [hunk 2]
```diff
@@ -105,6 +105,9 @@ let subAudioChunkLength = 0; // length of a sub audio chunk
 let subText = "";
 let speechToText = "";
 
+let time_to_first_token = 0; // TTFT
```
Comment (fdwr) [inline demos/whisper-base/main.js:108]:
```suggestion
let timeToFirstToken = 0; // TTFT
```

these_two_use_snake_case🐍 when theOthersAboveAndBelowUseCamelCase🐪. (yes, whisper.js is aberrant and can do its own thing, but it's good to be consistent within a given file main.js)

```
let maxUnprocessedAudioLength = 0;
let maxProcessAudioBufferLength = 0;
let accumulateSubChunks = false;
let silenceAudioCounter = 0;
...
const blacklistTags = [
```

**Review by fdwr — APPROVED** [review]
Looks informative. One comment.

File: demos/whisper-base/main.js [hunk 3]
```diff
@@ -198,13 +205,17 @@ async function process_audio(audio, starttime, idx, pos) {
     // done with audio buffer
     const processing_time = (performance.now() - starttime) / 1000;
     const total = audio.length / kSampleRate;
+    const token_per_sec = (numTokens - 1) / (processing_time - timeToFirstToken / 1000);
```
Comment (fdwr) [inline demos/whisper-base/main.js:208]:
```suggestion
    const tokensPerSecond = (numTokens - 1) / (processing_time - timeToFirstToken / 1000);
```

**Review by fdwr — APPROVED** [review]
👍

**Comment by fdwr** [comment]
FYI @Adele101. Merging.

**Pull request merged**